homeassistant/components/peblar/coordinator.py

```diff
@@ -59,6 +64,49 @@ class PeblarData:
     system: PeblarSystem
 
 
+def _coordinator_exception_handler[
+    _DataUpdateCoordinatorT: PeblarDataUpdateCoordinator
+    | PeblarVersionDataUpdateCoordinator
+    | PeblarUserConfigurationDataUpdateCoordinator,
+    **_P,
+](
+    func: Callable[Concatenate[_DataUpdateCoordinatorT, _P], Coroutine[Any, Any, Any]],
+) -> Callable[Concatenate[_DataUpdateCoordinatorT, _P], Coroutine[Any, Any, Any]]:
+    """Handle exceptions within the update handler of a coordinator."""
+
+    async def handler(
+        self: _DataUpdateCoordinatorT, *args: _P.args, **kwargs: _P.kwargs
+    ) -> Any:
+        try:
+            return await func(self, *args, **kwargs)
```

Comment:
Why don't you create a base coordinator and have the try catch wrapped around a function for the others to implement?

Reply:
🤷  Same same but different :)